src/components/elements/inputs/file-input.js

```diff
@@ -0,0 +1,130 @@
+import { Component } from 'react'
+import cx from 'classnames'
+
+export default class FileInput extends Component {
+
+  render() {
+    const opts = {}
+    if (this.props.property.isArray) {
+      opts['multiple'] = 'multiple'
+    }
+    return (
+
+      <div className={cx(this.props.className, 'ct-file-input') }>
+        <input
+          {...opts}
+          id={this.props.property.name}
+          onChange={e => this.handleFiles(e).then(file => this.props.onSave(file))}
+          type={'file'}
+          style={{ display: 'none' }}
+          ref={inputElement => this.fileInput = inputElement}
+          />
+        <button
+          className={'ct-action'}
+          title={this.props.property.title}
+          disabled={this.props.property.disabled}
+          onClick={() => this.fileInput.click() }>
+          {this.renderTitle()}
+        </button>
+        <span className={'ct-selected-information'}>{this.renderFiles(this.props.value) }</span>
```

Comment:
Tried to change, but I'm sorry I have to say I feel like [this](https://www.youtube.com/watch?v=S82HH5bY5PE)...